src/lib/registration/controllers/package_search.rb

```diff
@@ -0,0 +1,189 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "registration/package_search"
+
+Yast.import "Popup"
+Yast.import "HTML"
+
+module Registration
+  module Controllers
+    # Implements the actions and keeps the state for the package search feature
+    class PackageSearch
+      include Yast::I18n
+
+      # @return [Array<RemotePackage>] List of selected packages
+      attr_reader :selected_packages
+
+      # Constructor
+      def initialize
+        textdomain "registration"
+        @selected_packages = []
+      end
+
+      # Returns the list of the current search
+      #
+      # @return [Array<RemotePackage>] List of found packages
+      def packages
+        @search ? @search.packages : []
+      end
+
+      # Performs a package search
+      #
+      # @param text [String] Term to search for
+      def search(text)
```

Comment:
To be honest I do not like this search with such huge side effect. and also with strict call order, as packages depends on search called in advance.
In general I do not like much mixing of holder for selection together with search filtering. I would keep it separate.